src/operation.go

```diff
@@ -54,11 +55,21 @@ func (d *operationExecutor) ExecOperation(ctx context.Context, cfg config.Config
 		return err
 	}
 
-	if cfg.Infrastructures["my-plugin"] != nil {
-		for _, k := range cfg.Infrastructures["my-plugin"].Keys() {
-			log.Printf("configuration key: %s", k)
-		}
-		log.Printf("Secret key: %q", cfg.Infrastructures["my-plugin"].GetStringOrDefault("test", "not found!"))
+	var locationProps config.DynamicMap
+	locationMgr, err := locations.GetManager(cfg)
+	if err != nil {
+		return err
+	}
+
+	locationProps, err = locationMgr.GetLocationProperties("my-plugin-location", "my-plugin-infra")
```

Comment:
Maybe here, as `locationMgr.GetLocationProperties()` has already been seen in the the delegate, we could use here another way to ge the location properties with `locationMgr.GetLocationPropertiesForNode()`:

```
locationProps, err = locationMgr.GetLocationPropertiesForNode(deploymentID, nodeName, "my-plugin-infra")
```

Reply:
Yes, sounds good !